src/ipc-handlers.ts

```diff
@@ -1267,6 +1268,17 @@ export async function isCATrusted(): Promise< boolean > {
 	return isRootCATrusted();
 }
 
+export async function trustCertificate(): Promise< boolean > {
+	const platform = process.platform;
+	if ( platform === 'win32' ) {
+		return trustRootCA();
+	}
+
+	void openCertificateDialog();
+
+	return true;
+}
```

Comment:
```suggestion
export async function trustCertificate( event: IpcMainInvokeEvent ): Promise< void > {
	if ( process.platform === 'win32' ) {
		const result = await trustRootCA();

		if ( ! result ) {
			await showMessageBox( event, {
				type: 'error',
				message: __( 'Certificate Trust Failed' ),
				detail: __(
					'Studio was unable to trust the certificate automatically. You may need to trust it manually using certificate manager.'
				),
			} );
		}
	}

	await openCertificateDialog();
}
```
I'd advise us to keep the logic for displaying the message box in the IPC handler function, and keep the component focused on calling this single IPC function.